Automattic/wp-calypso
# Add Jetpack Settings Host Detection

#### Changes proposed in this Pull Request

**Motivation**: Add the host detection step from the SSH credentials host specific instructions project ( p1HpG7-9Nd-p2 ).

* Add the host detection/guess/selection step from the project
* Support new WPCOM Host Guess API
* Add new routes for host detection and credentials step in the Jetpack Cloud settings module
* Rejigger file structure in the Jetpack Cloud settings module
* Minor update to StepProgress Component to accept translation results as well as strings

![Untitled 9@0 5x](https://user-images.githubusercontent.com/2810519/92269369-e4d7a980-ee98-11ea-87d9-fa64823fcd6b.png)


#### Testing instructions

1. boot branch w/ `env ENABLE_FEATURES=jetpack/server-credentials-advanced-flow yarn run start-jetpack-cloud`
2. Navigate to "Settings" for a site.
3. Verify your screen matches "Host Selection w/out guess" above"
4. Chose a host, verify it now match "Credentials Placeholder"
5. Apply 49131-code & sandbox `public-api.wordpress.com`
6. redo steps 2-4, this time expecting "Host Selection w/ guess" if your host matches one on the list
